test/gtest/uct/test_pd.cc

```diff
@@ -83,20 +91,101 @@ std::vector<std::string> test_pd::enum_mds(const std::string& mdc_name) {
 
 test_pd::test_pd()
 {
+    skip_test = 0;
+    support_odp = 0;
+    tl = std::tr1::get<0>(GetParam());
+    reg_method = std::tr1::get<1>(GetParam());
     UCS_TEST_CREATE_HANDLE(uct_md_config_t*, m_md_config,
                            (void (*)(uct_md_config_t*))uct_config_release,
-                           uct_md_config_read, GetParam().c_str(), NULL, NULL);
+                           uct_md_config_read, tl.c_str(), NULL, NULL);
 }
 
 void test_pd::init()
 {
+    std::string tl;
+
+    tl = std::tr1::get<0>(GetParam());
+    reg_method = std::tr1::get<1>(GetParam());
+    struct ibv_device **dev_list;
+    struct ibv_device **orig_dev_list;
+    int num_devices;
+
+    skip_test = 0;
+
+    support_odp = 0;
+#if HAVE_STRUCT_IBV_EXP_DEVICE_ATTR_ODP_CAPS
+    if (reg_method == "odp") {
+        orig_dev_list = dev_list = ibv_get_device_list(&num_devices);
+        if (!dev_list) {
+            UCS_TEST_ABORT("Failed to get the device list.");
+        }
+
+        while (*dev_list) {
```

Comment:
please query for odp support from the IB MD structure (for example from uct_ib_device_t)